fire/cli/ts/statistics.py

```diff
@@ -0,0 +1,337 @@
+from fire.cli.ts.dataloader import base_data, ENH_data, grid_data
+from dataclasses import dataclass
+
+import datetime
+import numpy as np
+import warnings
+from sklearn.linear_model import LinearRegression
+from pyproj import Proj
+from scipy.stats import t
+from scipy.interpolate import interp2d
+from scipy.interpolate import RegularGridInterpolator
+
+from typing import List
+import numpy.typing as npt
+
+
+@dataclass
+class statvar:
+    x: npt.NDArray[datetime.date] = None
+    y: npt.NDArray[np.float64] = None
+    X: npt.NDArray[np.float64] = None
+    Y: npt.NDArray[np.float64] = None
+    nr: int = None
+    N_full: int = None
+    N_binned: int = None
+    med: datetime.date = None
+    height_fit: npt.NDArray[np.float64] = None
+    r2: float = None
+    slope_fit: float = None
+    CI_L: float = None
+    CI_U: float = None
+    t_slope: float = None
+    CI_L_slope: float = None
+    CI_U_slope: float = None
+    var_pool: float = None
+    std_pool: float = None
+    z_crit: float = None
+    CI_L_pool: float = None
+    CI_U_pool: float = None
+    meh: float = None
+    u: float = None
+    uplift_corr: float = None
+    t_val: float = None
+    t_crit: float = None
+    t_std: float = None
+    delta_uplift: float = None
+    z_score: float = None
+
+
+binOutType = base_data, ENH_data
+
+
+def binning(obs: base_data, ENH: ENH_data, binsize: int) -> binOutType:
```

Comment:
`binsize` er ikke end vanvittigt god betegnelse, især når den er specificeret til at være en int: Det ser ud som om `binsize` markerer et fælles antal punkter for hver bin.

Den burde nok være en float (hvem siger at man ikke føler trang til at betragte æsker á halvandet døgn - eller fx en månefases tid?)

Hvad navnet angår: Her skal tænkes lidt!

Og ved nærmere eftertanke bliver jeg lidt bims her: Jeg kan ikke hurtigt gennemskue hvad der egentlig er definerende for hvad der hører til i en bin: Jeg går ud fra at binsize angiver det hul der skal være i en tidsrække før man afbryder og går til en ny bin. Men det kan jo føre til meget lange midlinger, hvis binsize er længere end afstanden mellem (de fleste) observationer?

Reply:
Se min kommentar ovenfor. Kan godt lave binsize som float men det vil sjældent være relevant at bruge halve dage i denne sammenhæng. Lyder som om du tænker på noget spektralanalyse hvilket kunne være super interessant for tidsserier med meget data, dog er det uden for hvad programmet kan (og skal kunne?) lige nu. :)